database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -274,26 +274,57 @@ private void writeToDataOut(DataOutput out, Schema.Field field) throws IOExcepti
     }
   }
 
-  protected void writeToDB(PreparedStatement stmt, @Nullable Schema.Field field, int fieldIndex) throws SQLException {
+  private void writeToDB(PreparedStatement stmt, @Nullable Schema.Field field, int fieldIndex) throws SQLException {
+    if (shouldWriteNullField(field, fieldIndex)) {
+      writeNullToDB(stmt, field, fieldIndex);
+    } else {
+      writeNonNullToDB(stmt, field, fieldIndex);
+    }
+  }
+
+  /**
+   * This method returns true in case a field can support writeNullToDB for the current field.
+   * By default, this method returns true when field or field value is set to null.
+   *
+   * @param field Field
+   * @param fieldIndex Index of the field
+   * @return true if null value of the field can be written to DB
+   */
+  protected boolean shouldWriteNullField(Schema.Field field, int fieldIndex) {
+    return (field == null || record.get(field.getName()) == null);
+  }
 
+  /**
+   * This method handle the null field and null field values, by internally using the PreparedStatement.setNull
+   * method. Any class requiring a custom handling to write null value for any type should override this method.
+   *
+   * @param stmt       PreparedStatement object for writing to db
+   * @param field      Current Field from record's schema
+   * @param fieldIndex Field index in the columnTypes
+   * @throws SQLException Exception while calling PreparedStatement.setNull
+   */
+  protected void writeNullToDB(PreparedStatement stmt,
+                               @Nullable Schema.Field field, int fieldIndex) throws SQLException {
     int sqlIndex = fieldIndex + 1;
     int sqlType = columnTypes.get(fieldIndex).getType();
-    if (field == null) {
-      // Some of the fields can be absent in the record
-      stmt.setNull(sqlIndex, sqlType);
-      return;
-    }
+    stmt.setNull(sqlIndex, sqlType);
+  }
 
+  /**
+   * Write Non null values to DB.
+   *
+   * @param stmt          PreparedStatement object for writing to db
+   * @param field         Current Field from record's schema
+   * @param fieldIndex    Field index in the columnTypes
+   * @throws SQLException Exception while calling PreparedStatement.setNull
+   */
+  protected void writeNonNullToDB(PreparedStatement stmt,
+                                  @Nullable Schema.Field field, int fieldIndex) throws SQLException {
+
+    int sqlIndex = fieldIndex + 1;
     String fieldName = field.getName();
     Schema fieldSchema = getNonNullableSchema(field);
```

Comment:
it would be good to do this in the caller and then pass in the fieldName and Schema as arguments instead of the Schema.Field.

That way anything that implements this method doesn't need to do the getNonNullable logic, which can be pretty easy to forget to do.